scripts/install.js

```diff
@@ -92,37 +92,47 @@ function download(url, dest, cb) {
  */
 
 function checkAndDownloadBinary() {
-  if (sass.hasBinary(sass.getBinaryPath())) {
+  if (process.env.SKIP_SASS_BINARY_DOWNLOAD_FOR_CI) {
+    console.log('Skipping downloading binaries on CI builds');
     return;
   }
+  var binaryPath = sass.getBinaryPath();
 
-  mkdir(path.dirname(sass.getBinaryPath()), function(err) {
+  if (sass.hasBinary(binaryPath)) {
+    return;
+  }
+
+  mkdir(path.dirname(binaryPath), function(err) {
     if (err) {
       console.error(err);
       return;
     }
 
-    download(sass.getBinaryUrl(), sass.getBinaryPath(), function(err) {
+    var binaryName = sass.getBinaryName(),
+      cachePath = sass.getBinaryCachePath(),
+      cacheBinary = sass.getCachedBinary();
+
+    if (cacheBinary) {
+      console.log('Found existing binary in ' + cacheBinary);
+      fs.createReadStream(cacheBinary).pipe(fs.createWriteStream(binaryPath));
+      return;
+    }
+
+    cacheBinary = path.join(cachePath, binaryName);
+
+    download(sass.getBinaryUrl(), cacheBinary, function(err) {
       if (err) {
         console.error(err);
         return;
       }
 
-      console.log('Binary downloaded and installed at', sass.getBinaryPath());
+      console.log('Binary downloaded to ' + cacheBinary);
+      fs.createReadStream(cacheBinary).pipe(fs.createWriteStream(binaryPath));
```

Comment:
The worse case here is that we risk copying from the vendor dir to the vendor dir. Could be problematic.